src/commands.ts

```diff
@@ -23,35 +24,50 @@ import * as vscode from 'vscode';
  *   https://code.visualstudio.com/api/extension-guides/command
  */
 
-/**
+// flags to indicating whether a resolve or update is in progress
+var resolveRunning = false;
+var updateRunning = false;
+
+ /**
  * Contains the commands defined in this extension.
- */
+ */ 
 const commands = {
 
     /**
      * Executes a {@link vscode.Task task} to resolve this package's dependencies.
      */
     async resolveDependencies() {
+        // return if running resolve or update already
+        if (resolveRunning || updateRunning) { return; }
+        resolveRunning = true;
+
         let tasks = await vscode.tasks.fetchTasks();
         let task = tasks.find(task =>
             task.definition.command === 'swift' &&
             task.definition.args[0] === 'package' &&
             task.definition.args[1] === 'resolve'
         )!;
-        vscode.tasks.executeTask(task);
+        await executeTaskAndWait(task);
+
+        resolveRunning = false;
     },
 
     /**
      * Executes a {@link vscode.Task task} to update this package's dependencies.
      */
     async updateDependencies() {
+        if (updateRunning) { return; }
```

Comment:
Do we want to allow an update if a resolve is already running? That might cause issues but I think it's probably ok

Reply:
We probably need better management of all the `swift` calls. Maybe a queue to avoid calls happening concurrently. But also ways to skip unnecessary calls. But we should leave that until a later PR.